packages/flutter/lib/src/rendering/object.dart

```diff
@@ -678,6 +682,40 @@ class PaintingContext extends ClipContext {
     return layer;
   }
 
+  /// Blend further painting with an alpha value.
+  ///
+  /// {@macro flutter.rendering.PaintingContext.pushClipRect.needsCompositing}
+  ///
+  /// The `offset` argument indicates an offset to apply to all the children
+  /// (the rendering created by `painter`).
+  ///
+  /// The `alpha` argument is the alpha value to use when blending the painting
+  /// done by `painter`. An alpha value of 0 means the painting is fully
+  /// transparent and an alpha value of 255 means the painting is fully opaque.
+  ///
+  /// The `painter` callback will be called while the `alpha` is applied. It
+  /// is called synchronously during the call to [pushOpacityLayer].
+  ///
+  /// {@macro flutter.rendering.PaintingContext.pushClipRect.oldLayer}
+  OpacityLayer? pushOpacityLayer(bool needsCompositing, Size size, Offset offset, int alpha, PaintingContextCallback painter, { OpacityLayer? oldLayer }) {
```

Comment:
Yeah I don't know what to do for naming, since the old name is taken

Reply:
Doesn't pushTransform do similar things? It may or may not push a layer.

Reply:
> Doesn't pushTransform do similar things? It may or may not push a layer.

Yeah, that's why it isn't called `pushTransformLayer` :) Ideally, we call this `pushOpacity`, but that name is already taken by the old method.